src/Cosi-BLS/transaction.lisp

```diff
@@ -67,26 +86,40 @@ It should be stored in the wallet"))
   (pedersen-commitment (txin-prf utx)))
 
 (defmethod pedersen-commitment ((utx uncloaked-txin))
-  (ed-nth-pt (txin-amt utx)))
+  (ed-decompress-pt (txin-prf utx)))
 
 (defmethod pedersen-commitment ((utx txout))
   (pedersen-commitment (txout-prf utx)))
 
 (defmethod pedersen-commitment ((utx uncloaked-txout))
-  (ed-nth-pt (uncloaked-txout-amt utx)))
+  (ed-decompress-pt (txout-prf utx)))
 
 (defmethod make-hashlock ((prf range-proofs:range-proof) (pkey pbc:public-key))
   "We are often given a long Bulletproof, and we need the to hash on
 the simple commitment to the uncloaked value"
   (hash:hash/256 (pedersen-commitment prf) pkey))
 
-(defmethod make-hashlock ((amt integer) (pkey pbc:public-key))
-  (hash:hash/256 amt pkey))
+(defmethod make-hashlock (arg (pkey pbc:public-key))
+  (hash:hash/256 arg pkey))
+
+(defun make-admin-commit (amt &optional (gamma (random-between 1 *ed-r*)))
+  (values
+   (range-proofs:simple-commit (range-proofs:hpt) gamma amt)
+   gamma))
+
+(defun check-amt (amt)
+  (and (not (minusp amt))
+       (< amt #.(ash 1 64))))
```

Comment:
@dbmcclain:  Where does the maximum amount of `(ash 1 64)` come from?  The simple need to have an upper bound?  Or is this related to some sort of serialization constraint?  Or is this the maximum that the range proofs can contain?